### PR TITLE
feat: ExecutionMode control for Flexible Processing

### LINKED ISSUE #2532: FlexibleProcessing runs in GameStudio always
**Release Type**: Official Release

**Version**: 4.2.0.2267

**Platform(s)**: Windows

**Describe the bug**
It seems like the bug is related to the Editor trying to run the `FlexibleProcessor` for `ISimulation` components. 

**To Reproduce**
1. Open the sample from https://github.com/stride3d/stride/tree/master/sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Demo in GameStudio
2. Open Scene number 9
3. witness massive slowdowns and an initial error message trying to use the bepu sim

**Expected behavior**
`ISimulation` components should not run in editor unless specified.

**Log and callstacks**
```
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at Stride.BepuPhysics.Demo.Components.Utils.Spawner.Spawn(Vector3 position, Vector3 Impulse, Vector3 ImpulsePos) in C:\dev\stride-dev\BepuDemo\BepuDemo\Stride.BepuPhysics.Demo\Components\Utils\_Spawner.cs:line 27
   at Stride.BepuPhysics.Demo.Components.Utils.SpawnerComponent.SimulationUpdate(BepuSimulation simulation, Single timeStep) in C:\dev\stride-dev\BepuDemo\BepuDemo\Stride.BepuPhysics.Demo\Components\Utils\SpawnerComponent.cs:line 52
   at Stride.BepuPhysics.BepuSimulation.Elider.Handler`1.SimulationUpdate(BepuSimulation sim, Single deltaTime) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\engine\Stride.BepuPhysics\Stride.BepuPhysics\BepuSimulation.cs:line 915
   at Stride.BepuPhysics.BepuSimulation.Elider.SimulationUpdate(Dictionary`2 handlers, BepuSimulation sim, Single deltaTime) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\engine\Stride.BepuPhysics\Stride.BepuPhysics\BepuSimulation.cs:line 897
   at Stride.BepuPhysics.BepuSimulation.Update(TimeSpan elapsed) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\engine\Stride.BepuPhysics\Stride.BepuPhysics\BepuSimulation.cs:line 718
   at Stride.BepuPhysics.Systems.PhysicsGameSystem.Update(GameTime time) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\engine\Stride.BepuPhysics\Stride.BepuPhysics\Systems\PhysicsGameSystem.cs:line 33
   at Stride.Games.GameSystemCollection.Update(GameTime gameTime)
   at Stride.Editor.EditorGame.Game.EditorServiceGame.Update(GameTime gameTime) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\editor\Stride.Editor\EditorGame\Game\EditorServiceGame.cs:line 180
```

### PR DESCRIPTION
**Pull request by Eideren**
# PR Details
See title

## Related Issue
Fix #2532

## Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] **I have built and run the editor to try this change out.**

### CODE REVIEW

File: sources/engine/Stride.Engine/Engine/FlexibleProcessing/IProcessorBase.cs [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
 
 using Stride.Core;
+using Stride.Engine.Design;
 
 namespace Stride.Engine.FlexibleProcessing
 {
     public interface IProcessorBase
     {
+        /// <summary> The logic using this value expects it to be constant at runtime </summary>
+        ExecutionMode ExecutionMode => ExecutionMode.Runtime;
```
Comment (Doprez) [inline sources/engine/Stride.Engine/Engine/FlexibleProcessing/IProcessorBase.cs:12]: Do we want the default to act like a regular `EntityProcessor` for the sake of consistency?

**Review by Doprez — COMMENTED** [review]
One question but functionally this looks good to me.

File: sources/engine/Stride.Engine/Engine/FlexibleProcessing/IProcessorBase.cs (hunk 1 above)
Comment (Eideren) [reply sources/engine/Stride.Engine/Engine/FlexibleProcessing/IProcessorBase.cs:12]: I think most users would actually expect their processors to only run at runtime out of the box, like script do. I understand that it's a bit counter intuitive when you come from how `EntityProcessor` work out of the box, which is why I wanted to change that as well but that's a far bigger breaking change

Comment (Doprez) [reply sources/engine/Stride.Engine/Engine/FlexibleProcessing/IProcessorBase.cs:12]: With the amount of times I have crashed the editor due to me forgetting to add `ExecutionMode` I fully agree with that logic. 

Sounds good to me then.

**Review by Doprez — APPROVED** [review]

**Pull request merged**